mr/src/test/java/org/apache/iceberg/mr/hive/serde/objectinspector/TestIcebergRecordObjectInspector.java

```diff
@@ -61,4 +61,17 @@ public void testIcebergRecordObjectInspector() {
     Assert.assertEquals(innerRecord.get(0), innerSoi.getStructFieldData(innerData, stringField));
   }
 
+  @Test
+  public void testIcebergRecordObjectInspectorWithRowNull() {
+    Schema schema = new Schema(
+            required(1, "integer_field", Types.IntegerType.get()),
+            required(2, "struct_field", Types.StructType.of(
+                    Types.NestedField.required(3, "string_field", Types.StringType.get())))
+    );
+    StructObjectInspector soi = (StructObjectInspector) IcebergObjectInspector.create(schema);
+    Assert.assertNull(soi.getStructFieldsDataAsList(null));
+    StructField integerField = soi.getStructFieldRef("integer_field");
+    Assert.assertNull(soi.getStructFieldData(null, integerField));
```

Comment:
Would it be possible to stick a null column into an e2e Hive job, to ensure that reading it doesn't cause issues? This is just testing that the object inspector returns null appropriately. But it doesn't test if other code (which previously wasn't getting null) will hit any NPEs or anything.

If that test is too complicated to add in somehow, probably not the end of the world 🙂

Reply:
`TestHiveIcebergStorageHandlerWithEngine` might be a place to test out the issue E2E

Reply:
@kbendick  @pvary, i have an integration test in TestHiveIcebergStorageHandlerWithEngine and edit the commit message
Thanks for the review